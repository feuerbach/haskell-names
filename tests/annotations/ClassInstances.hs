module ClassInstances where

data D a = D DataType a

class C a where
    wiggle :: a -> a
    woe :: a
    method1 :: a
    ($$$) :: a -> a -> a

instance (C a) => C (D a) where
    wiggle (D b a) = D b (f a)
    woe = D Constructor1 woe
    method1 = D Constructor1 woe
    f $$$ x = ($$$) f x

f :: (C a) => a -> a
f x = wiggle x
