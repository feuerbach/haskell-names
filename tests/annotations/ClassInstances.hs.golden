D        at  3:6 is a type or class defined here
a        at  3:8 is none
D        at 3:12 is a value bound here
DataType at 3:14 is a global data type, Prelude.DataType
DataType at 3:14 is a global data type, Prelude.DataType
a        at 3:23 is none
C        at  5:7 is a type or class defined here
a        at  5:9 is none
wiggle   at  6:5 is a value bound here
a        at 6:15 is none
a        at 6:20 is none
woe      at  7:5 is a value bound here
a        at 7:12 is none
method1  at  8:5 is a value bound here
a        at 8:16 is none
$$$      at  9:5 is a value bound here
a        at 9:14 is none
a        at 9:19 is none
a        at 9:24 is none
C        at 11:11 is a global type class, ClassInstances.C
C        at 11:11 is a global type class, ClassInstances.C
a        at 11:13 is none
C        at 11:19 is a global type class, ClassInstances.C
C        at 11:19 is a global type class, ClassInstances.C
D        at 11:22 is a global data type, ClassInstances.D
D        at 11:22 is a global data type, ClassInstances.D
a        at 11:24 is none
wiggle   at 12:5 is a global method, ClassInstances.wiggle
D        at 12:13 is a global constructor, ClassInstances.D
D        at 12:13 is a global constructor, ClassInstances.D
b        at 12:15 is a value bound here
a        at 12:17 is a value bound here
D        at 12:22 is a global constructor, ClassInstances.D
D        at 12:22 is a global constructor, ClassInstances.D
b        at 12:24 is a local value defined at 12:15
b        at 12:24 is a local value defined at 12:15
f        at 12:27 is a global value, ClassInstances.f
f        at 12:27 is a global value, ClassInstances.f
a        at 12:29 is a local value defined at 12:17
a        at 12:29 is a local value defined at 12:17
woe      at 13:5 is a global method, ClassInstances.woe
D        at 13:11 is a global constructor, ClassInstances.D
D        at 13:11 is a global constructor, ClassInstances.D
Constructor1 at 13:13 is a global constructor, Prelude.Constructor1
Constructor1 at 13:13 is a global constructor, Prelude.Constructor1
woe      at 13:26 is a global method, ClassInstances.woe
woe      at 13:26 is a global method, ClassInstances.woe
method1  at 14:5 is a global method, ClassInstances.method1
D        at 14:15 is a global constructor, ClassInstances.D
D        at 14:15 is a global constructor, ClassInstances.D
Constructor1 at 14:17 is a global constructor, Prelude.Constructor1
Constructor1 at 14:17 is a global constructor, Prelude.Constructor1
woe      at 14:30 is a global method, ClassInstances.woe
woe      at 14:30 is a global method, ClassInstances.woe
f        at 15:5 is a value bound here
$$$      at 15:7 is a global method, ClassInstances.($$$)
x        at 15:11 is a value bound here
$$$      at 15:15 is a global method, ClassInstances.($$$)
$$$      at 15:15 is a global method, ClassInstances.($$$)
f        at 15:21 is a local value defined at 15:5
f        at 15:21 is a local value defined at 15:5
x        at 15:23 is a local value defined at 15:11
x        at 15:23 is a local value defined at 15:11
f        at 17:1 is a global value, ClassInstances.f
C        at 17:7 is a global type class, ClassInstances.C
C        at 17:7 is a global type class, ClassInstances.C
a        at 17:9 is none
a        at 17:15 is none
a        at 17:20 is none
f        at 18:1 is a value bound here
x        at 18:3 is a value bound here
wiggle   at 18:7 is a global method, ClassInstances.wiggle
wiggle   at 18:7 is a global method, ClassInstances.wiggle
x        at 18:14 is a local value defined at 18:3
x        at 18:14 is a local value defined at 18:3
